packages/cosmos_ui_components/lib/components/cosmos_switch.dart

```diff
@@ -13,26 +14,89 @@ class CosmosSwitch extends StatelessWidget {
   final bool checked;
   final ValueChanged<bool>? onChanged;
 
+  @override
+  CosmosSwitchState createState() => CosmosSwitchState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(ObjectFlagProperty<ValueChanged<bool>?>.has('onChanged', onChanged))
+      ..add(DiagnosticsProperty<bool>('checked', checked));
+  }
+}
+
+class CosmosSwitchState extends State<CosmosSwitch> with SingleTickerProviderStateMixin {
+  late Animation<Alignment> _circleAnimation;
+  late AnimationController _animationController;
+
+  @override
+  void initState() {
+    super.initState();
+    _animationController = AnimationController(vsync: this, duration: const Duration(milliseconds: 200));
+    _circleAnimation = AlignmentTween(
+      begin: widget.checked ? Alignment.centerRight : Alignment.centerLeft,
+      end: widget.checked ? Alignment.centerLeft : Alignment.centerRight,
+    ).animate(
+      CurvedAnimation(
+        parent: _animationController,
+        curve: Curves.linear,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () {
+        if (_animationController.isCompleted) {
+          _animationController.reverse();
+        } else {
+          _animationController.forward();
+        }
```

Comment:
this is logic inside build method. also, specifying ANimationController is too low level and requires a lot of boilerplate code and maintenance. you have widgets for such simple animations, for example `TweenAnimationBuilder`. Have a look at `expandable_container.dart` for an example how to use it